### PR TITLE
bugfix when doing photometry and some objects very very near zero ra dec

### PR DESCRIPTION
**Pull request by dokeeffe**
Found this when running against my HAT-P-18b data  https://twitter.com/BallyhouraStars/status/1114895128286642176

Some of the RA coordinates were almost zero and were in the temp file in scientific notation. The regex just needed a small tweak. Test added to replicate problem before fix.

### CODE REVIEW

File: test/test_methods.py [hunk 1]
```diff
@@ -149,6 +149,18 @@ def test_load_coordinates(self):
                 for coords, expected in zip(coordinates, objects):
                     self.assertEqual(coords, expected)
 
+    def test_load_coordinates_sci_notation(self):
+
+        # For some datasets that generate coords so close to zero that they end up in scientific notation
+
+        for _ in xrange(NITERS):
```
Comment (vterron) [inline test/test_methods.py]: We're always using the same value, so we don't need a for loop here.

File: test/test_methods.py [hunk 2]
```diff
@@ -149,6 +149,18 @@ def test_load_coordinates(self):
...
+            data = '7.9720694373e-05 44.6352243008'
+            with methods.tempinput(data) as path:
+                coordinates = methods.load_coordinates(path)
+                for coords in coordinates:
```
Comment (vterron) [inline test/test_methods.py]: The list returned by `load_coordinates()` contains a single element, so instead of a for loop:

```
coordinates = methods.load_coordinates(path)
self.assertEqual(len(coordinates), 1)
coords = coordinates[0]
self.assertEqual(coords, (7.9720694373e-05, 44.6352243008, None, None))
```

File: test/test_methods.py [hunk 3]
```diff
@@ -149,6 +149,18 @@ def test_load_coordinates(self):
...
+                    self.assertEqual(coords, (7.9720694373e-05, 44.6352243008, None, None))
```
Comment (vterron) [inline test/test_methods.py]: For extra points, you could add a separate test where the proper motions are not `None`.

File: test/test_methods.py [hunk 4]
```diff
@@ -149,6 +149,18 @@ def test_load_coordinates(self):
                 for coords, expected in zip(coordinates, objects):
                     self.assertEqual(coords, expected)
 
+    def test_load_coordinates_sci_notation(self):
```
Comment (vterron) [inline test/test_methods.py]: s/sci/scientific

File: test/test_methods.py (hunk 1 above)
Comment (dokeeffe) [reply test/test_methods.py]: Fixed

File: test/test_methods.py (hunk 2 above)
Comment (dokeeffe) [reply test/test_methods.py]: Done, Its returning a generator so I converted to a List

File: test/test_methods.py (hunk 3 above)
Comment (dokeeffe) [reply test/test_methods.py]: Done

File: test/test_methods.py [hunk 5]
```diff
@@ -149,6 +149,29 @@ def test_load_coordinates(self):
...
+        data = '7.9720694373e-05 44.6352243008'
+        with methods.tempinput(data) as path:
+            coordinates = methods.load_coordinates(path)
+            coords_list = list(coordinates)
+            self.assertEqual(len(coords_list), 1)
+            self.assertEqual(coords_list[0], (7.9720694373e-05, 44.6352243008, None, None))
+
+    def test_load_coordinates_scientific_notation_with_propper_motion(self):
+
+        # For some datasets that generate coords so close to zero that they end up in scientific notation
```
Comment (vterron) [inline test/test_methods.py]: This is the same comment as above, so I don't think we need it.

File: test/test_methods.py [hunk 6]
```diff
@@ -149,6 +149,29 @@ def test_load_coordinates(self):
...
+    def test_load_coordinates_scientific_notation(self):
+
+        # For some datasets that generate coords so close to zero that they end up in scientific notation
+
+        data = '7.9720694373e-05 44.6352243008'
+        with methods.tempinput(data) as path:
+            coordinates = methods.load_coordinates(path)
+            coords_list = list(coordinates)
+            self.assertEqual(len(coords_list), 1)
+            self.assertEqual(coords_list[0], (7.9720694373e-05, 44.6352243008, None, None))
```
Comment (vterron) [inline test/test_methods.py]: I realize now that we're comparing floating-point numbers, so `assertEqual()` isn't the best practice and may break at some point. Instead, what if we do:

```
ra, dec, pm_ra, pm_dec = coords_list[0]
self.assertAlmostEqual(ra, 7.9720694373e-05)
self.assertAlmostEqual(dec, 44.6352243008)
...
```

Same for the tests below (and feel free to fix the existing one above).

File: test/test_methods.py (hunk 4 above)
Comment (dokeeffe) [reply test/test_methods.py]: Sorry, I didn't understand :-) Do you mean rename the function by removing 'sci'?
Maybe calling it `test_load_coordinates`?

File: test/test_methods.py (hunk 6 above)
Comment (dokeeffe) [reply test/test_methods.py]: good point I'll change that

File: test/test_methods.py (hunk 4 above)
Comment (vterron) [reply test/test_methods.py]: Apologies for not being clear: I meant that we should call it 
`test_load_coordinates_scientific_notation`.

(`s/.../...` [comes from sed](https://www.gnu.org/software/sed/manual/html_node/The-_0022s_0022-Command.html)).

File: test/test_methods.py (hunk 6 above)
Comment (dokeeffe) [reply test/test_methods.py]: Done

File: test/test_methods.py (hunk 5 above)
Comment (dokeeffe) [reply test/test_methods.py]: removed

File: test/test_methods.py (hunk 4 above)
Comment (dokeeffe) [reply test/test_methods.py]: Ah ok yes, I renamed it :-)

**Comment by dokeeffe** [comment]
Victor, I think all code review comments are addressed. 
Anything else to improve, let me know.

I think only you can merge it, I dont have the merge button :-)

**Comment by vterron** [comment]
Merged. Thanks again!
